contracts/strategies/stkCvxCrv/harvester/Harvester.sol

```diff
@@ -0,0 +1,232 @@
+            : 0;
+        cvxEthSwap.exchange_underlying{value: 0}(1, 0, _amount, _minAmountOut);
+    }
+
+    /// @notice Compute a min amount of ETH based on pool oracle / vprice for 3crv
+    /// @param _amount - amount to swap
+    /// @return min acceptable amount of ETH
+    function _calcMinAmountOutThreeCrvEth(uint256 _amount)
+        internal
+        returns (uint256)
+    {
+        /// assume peg/balance for 3crv pricing in USDT
+        uint256 _virtualPrice = tripool.get_virtual_price();
+        uint256 _usdtAmount = (_amount * _virtualPrice) / 1e18;
+        /// get ETH price in USDT from tricrypto
+        uint256 _ethUsdPrice = tricrypto.price_oracle(1);
+        uint256 _amountEth = ((_usdtAmount * 1e18) / _ethUsdPrice);
+        return ((_amountEth * allowedSlippage) / DECIMALS);
+    }
+
+    function _threeCrvToEth(uint256 _amount) internal {
+        uint256 _minAmountOut = useOracle
+            ? _calcMinAmountOutThreeCrvEth(_amount)
+            : 0;
+        tripool.remove_liquidity_one_coin(_amount, 2, 0);
+        uint256 _usdtBalance = IERC20(USDT_TOKEN).balanceOf(address(this));
+        if (_usdtBalance > 0) {
+            tricrypto.exchange(0, 2, _usdtBalance, _minAmountOut, true);
+        }
+    }
+
+    /// @notice Compute a min amount of CRV based on pool oracle for ETH
+    /// @param _amount - amount to swap
+    /// @return min acceptable amount of CRV
+    function _calcMinAmountOutEthCrv(uint256 _amount)
+        internal
+        returns (uint256)
+    {
+        uint256 _crvEthPrice = crvEthSwap.price_oracle();
+        uint256 _amountCrvPrice = ((_amount * 1e18) / _crvEthPrice);
+        return ((_amountCrvPrice * allowedSlippage) / DECIMALS);
+    }
+
+    function _ethToCrv(uint256 _amount) internal {
+        if (_amount > 0) {
+            uint256 _minAmountOut = useOracle
+                ? _calcMinAmountOutEthCrv(_amount)
+                : 0;
+            crvEthSwap.exchange_underlying{value: _amount}(
+                0,
+                1,
+                _amount,
+                _minAmountOut
+            );
+        }
+    }
+
+    function _crvToCvxCrv(uint256 _amount) internal returns (uint256) {
+        return crvCvxCrvSwap.exchange(0, 1, _amount, 0, address(this));
+    }
+
+    function processRewards(bool _forceLock)
+        external
+        onlyStrategy
+        returns (uint256)
+    {
+        uint256 _cvxBalance = IERC20(CVX_TOKEN).balanceOf(address(this));
+        if (_cvxBalance > 0) {
+            _cvxToEth(_cvxBalance);
+        }
+        uint256 _threeCrvBalance = IERC20(THREECRV_TOKEN).balanceOf(
+            address(this)
+        );
+        if (_threeCrvBalance > 0) {
+            _threeCrvToEth(_threeCrvBalance);
+        }
+        _ethToCrv(address(this).balance);
+        uint256 _crvBalance = IERC20(CRV_TOKEN).balanceOf(address(this));
+        if (_crvBalance > 0) {
+            // no oracle yet on the pool so we use get_dy
```

Comment:
sadge

Reply:
worst case if there's a sandwich harvest gets locked instead of swapping with a discount